src/main/java/hudson/plugins/clover/CloverPublisher.java

```diff
@@ -136,10 +149,29 @@ static File getCloverXmlReport(Run<?, ?> build) {
         return new File(build.getRootDir(), "clover.xml");
     }
 
+    /**
+     * Gets the directory where the Clover Report is stored for the given build with a specific reportId.
+     */
+    static File getCloverXmlReport(Run<?, ?> build, int reportId) {
+        String fileName = (reportId <= 0) 
+            ? "clover.xml" 
+            : "clover-" + reportId + ".xml";
```

Comment:
You have this logic twice. Here in getCloverXmlReport and in the copyXmlReport. Extract this to a helper static method.